src/python/build_management/build_manager.py

```diff
@@ -711,6 +713,57 @@ def setup(self):
 
     self._setup_application_path(build_update=build_update)
     self._post_setup_success(update_revision=build_update)
+
+    return True
+
+
+class FuchsiaBuild(Build):
+  """Represents a Fuchsia build."""
+
+  def __init__(self, base_build_dir, revision, target_weights=None):
+    super(FuchsiaBuild, self).__init__(base_build_dir, revision)
+    self.base_build_dir = ''
+    self.build_url = ''
+    self.build_dir_name = ''
+    self._build_dir = ''
+    self.target_weights = target_weights
+    self.revision = 0
+
+  @property
+  def build_dir(self):
+    return self._build_dir
+
+  def setup(self):
+    logs.log('Retrieving build %d.' % self.revision)
+
+    # Bucket for QEMU resources.
+    fuchsia_resources_dir = fuchsia.device.initialize_resources_dir()
+    environment.set_value('FUCHSIA_RESOURCES_DIR', fuchsia_resources_dir)
```

Comment:
I think now that we're trying to get this to fit the concept of a build, we should be storing these in the build directory instead.

In fact, making this fit as a "RegularBuild" (e.g. by subclassing that) will be even  better, because it comes with all the benefits of a proper build on CF, and should just work when we have revision numbers for dealing with different builds. We'll have to do some minor refactoring to factor out the target selection functionality though, so that you can override it.

Regular builds are builds which expect the following format. 

```
gs://bucket/path/fuchsia-resources-([0-9.]+).zip
```

When you call setup_build these are all unpacked into a subdir of `BUILDS_DIR` with a consistent path.

Fine to leave for another CL, but please do leave a TODO.